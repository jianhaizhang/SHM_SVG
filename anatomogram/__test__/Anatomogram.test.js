import React from 'react'
import { shallow } from 'enzyme'

import Anatomogram from '../src/Anatomogram'

describe(`Anatomogram`, () => {

  test(`should render for a supported species`, () => {
    expect(shallow(<Anatomogram
      idsWithMarkup={[]}
      onClick={jest.fn()}
      onMouseOut={jest.fn()}
      onMouseOver={jest.fn()}
      species={`mus_musculus`}
    />)).toMatchSnapshot()
  })

  test(`should not render for an unsupported species`, () => {
    expect(shallow(<Anatomogram atlasUrl={``} species={`ovis_aries`} />).children()).toHaveLength(0)
  })

})
