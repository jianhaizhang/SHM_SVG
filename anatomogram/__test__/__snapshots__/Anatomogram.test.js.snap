// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anatomogram should render for a supported species 1`] = `
<div>
  <Switcher
    atlasUrl="https://www.ebi.ac.uk/gxa/"
    onChangeView={[Function]}
    selectedView="male"
    species="mus_musculus"
  />
  <AnatomogramSvg
    atlasUrl="https://www.ebi.ac.uk/gxa/"
    idsWithMarkup={Array []}
    onClick={[MockFunction]}
    onInjected={[Function]}
    onMouseOut={[MockFunction]}
    onMouseOver={[MockFunction]}
    selectedView="male"
    species="mus_musculus"
  />
</div>
`;
